src/Db/Adapter/UnifiedMigrationsTableStorage.php

```diff
@@ -0,0 +1,270 @@
+ * Uses a single `cake_migrations` table with a `plugin` column
+ * to track all migrations (app and plugins) in one place.
+ */
+class UnifiedMigrationsTableStorage
+{
+    /**
+     * The table name for unified migrations storage.
+     */
+    public const TABLE_NAME = 'cake_migrations';
+
+    /**
+     * Constructor
+     *
+     * @param \Migrations\Db\Adapter\AbstractAdapter $adapter The database adapter.
+     * @param string|null $plugin The plugin name (null for app migrations).
+     */
+    public function __construct(
+        protected AbstractAdapter $adapter,
+        protected ?string $plugin = null,
+    ) {
+    }
+
+    /**
+     * Gets all the migration versions for the current plugin context.
+     *
+     * @param array<string, string> $orderBy The order by clause.
+     * @return \Cake\Database\Query\SelectQuery
+     */
+    public function getVersions(array $orderBy): SelectQuery
+    {
+        $query = $this->adapter->getSelectBuilder();
+        $query->select('*')
+            ->from(self::TABLE_NAME)
+            ->orderBy($orderBy);
+
+        if ($this->plugin === null) {
+            $query->where(['plugin IS' => null]);
+        } else {
+            $query->where(['plugin' => $this->plugin]);
+        }
+
+        return $query;
+    }
+
+    /**
+     * Records that a migration was run in the database.
+     *
+     * @param \Migrations\MigrationInterface $migration Migration
+     * @param string $startTime Start time
+     * @param string $endTime End time
+     * @return void
+     */
+    public function recordUp(MigrationInterface $migration, string $startTime, string $endTime): void
+    {
+        $query = $this->adapter->getInsertBuilder();
+        $query->insert(['version', 'migration_name', 'plugin', 'start_time', 'end_time', 'breakpoint'])
+            ->into(self::TABLE_NAME)
+            ->values([
+                'version' => (string)$migration->getVersion(),
+                'migration_name' => substr($migration->getName(), 0, 100),
+                'plugin' => $this->plugin,
+                'start_time' => $startTime,
+                'end_time' => $endTime,
+                'breakpoint' => 0,
+            ]);
+        $this->adapter->executeQuery($query);
+    }
+
+    /**
+     * Removes the record of a migration having been run.
+     *
+     * @param \Migrations\MigrationInterface $migration Migration
+     * @return void
+     */
+    public function recordDown(MigrationInterface $migration): void
+    {
+        $query = $this->adapter->getDeleteBuilder();
+        $query->delete()
+            ->from(self::TABLE_NAME);
+
+        if ($this->plugin === null) {
+            $query->where([
+                'version' => (string)$migration->getVersion(),
+                'plugin IS' => null,
+            ]);
+        } else {
+            $query->where([
+                'version' => (string)$migration->getVersion(),
+                'plugin' => $this->plugin,
+            ]);
+        }
```

Comment:
```suggestion
        $query->where([
            'version IS' => (string)$migration->getVersion(),
            'plugin' => $this->plugin,
        ]);

```

This should work as well.